projecttacoma/measure-repository
# Add button for deleting draft artifacts

# Summary
Adds button on each draft resource card to delete it. When clicked, the draft artifact is removed from the database and is no longer reflected via the draft resource counts or the rendered resource cards.

## New behavior
For draft resources, the user now has the option to delete a draft resource from the UI. When the trash icon is clicked, a confirmation modal appears to confirm the deletion. When confirmed, the resource is deleted from the database, its resource card is no longer rendered, and the draft artifact count is refreshed to reflect the updated count.

## Code changes
* `server/trpc/routers/draft.ts` - add new trpc procedure for deleting a draft artifact
* `server/db/dbOperations.ts` - add new function to delete the selected resource from the mongo collection
* `components/ResourceInfoCard.tsx`
    * Add delete button to each card, for authoring only, that pulls up a confirmation modal when clicked
    * Adds deletion confirmation modal that appears when the trash icon is clicked (probably easier to view this change locally rather than looking at the Github code diff)
    * When confirmed, a success notification appears, the resource is deleted from the database, and the draft counts and rendered resource cards are refreshed. If an error occurs, an error notification appears
* `components/ConfirmationModal.tsx` - new confirmation modal component, designed to be reusable if we have other confirmation modal uses in the future
* `util/types/fhir.ts` - update the resource card’s `resourceType` type to be `ArtifactResourceType` instead of `string` (since the rendered resource cards will always be of type Library or Measure)

# Testing guidance
* Navigate to the authoring mode and click on one of the resource count buttons to bring you to the resource cards for a given resource type. Check that the trash icon is rendered on each card
* Click on the trash icon. Check that the confirmation modal appears
* Click on “Delete” from the confirmation modal. Check the following:
    * The resource’s card no longer appears on the page
    * The draft artifact button for the given resource type is decremented by 1
    * The artifact is not in the database. You can do this by viewing the resource type collection in Robo3T
* Go through the same process, but instead click on “Cancel” from the confirmation modal. Check that the resource has not been deleted, and the UI has not changed in any way.
* Navigate to the repository mode, and check that the service resource cards have not been altered in any way (i.e. they should not have the trash icon, and they should still have the “details” icon rather than the “edit” icon)
